users/src/main/java/io/crate/operation/auth/AuthenticationService.java

```diff
@@ -40,7 +47,24 @@
     private final Map<String, AuthenticationMethod> authMethodRegistry = new HashMap<>();
     private Map<String, Map<String, String>> hbaConf;
 
-    AuthenticationService() {
+    private Object HBASetting;
+    private final ClusterService clusterService;
+
+    public void setHBAConf(Object HBASetting) {
+        this.HBASetting = HBASetting;
+        updateHbaConfig(getHBAConfig(HBASetting));
+    }
+
+    @Inject
+    AuthenticationService(ClusterService clusterService, Settings settings) {
+        this.clusterService = clusterService;
+        HBASetting = AuthenticationSettings.AUTH_HOST_BASED_SETTING.setting().get(settings);
+        updateHbaConfig(getHBAConfig(HBASetting));
+        clusterService.getClusterSettings()
+            .addSettingsUpdateConsumer(
+                AuthenticationSettings.AUTH_HOST_BASED_SETTING
+                    .setting(), this::setHBAConf);
```

Comment:
`this::updateHbaConfig`

so you have 2 more methods:

```java
void updateHbaConfig(Settings hbaConf) {
    updateHbaConfig(convertHbaSettingsToHbaConf(hbaConf));
}

private void convertHbaSettingsToHbaConf(Settings hbaSettings) {
...
}
```


this will make it easy to test